api/src/main/java/org/openmrs/module/appointments/service/impl/AppointmentsServiceImpl.java

```diff
@@ -140,6 +168,10 @@ public Appointment getAppointmentByUuid(String uuid) {
 
     @Override
     public void changeStatus(Appointment appointment, String status, Date onDate) throws APIException{
+        if (!validateIfUserHasSelfOrAllAppointmentsAccess(appointment)) {
```

Comment:
is this enough? For example, a provider may be one of the providers, but he can not "MANAGE APPOINTMENT" and/or this appointment was not created by him. - in such case, is he allowed to change Status?

Reply:
We are not considering the creator of the appointment as of now as that would involve more changes in the product. If the provider is one of the providers of the appointment, he/she should be able to change the status of the appointment. Lets say, if there are multiple providers for an appointment and patient was attended by only one of those providers, the provider should be able to change the status of the appointment in such a case else there has to be some notification system to do the same.

Reply:
I am ok with that. Please mention this explicitly in the JIRA card. Saying out of scope for no - if a provider is one of the assignees in the appointment, then he can change status.